feat: Add hosts file import feature (#248)

Bastian130 opened the pull request:
# Description

> This PR implements the hosts file import feature requested in issue #248. It allows users to provide a custom hosts file during container creation/start, with automatic persistence across restarts.

# Related issues

> Fixes #248

Bastian130 committed:
feat: Add hosts file import feature (#248)

Dramelac commented on exegol/model/ContainerConfig.py:
```diff
@@ -1374,6 +1376,37 @@ def getShellEnvs(self) -> List[str]:
+                    if not line or line.startswith('#'):
+                        continue
+                    # Parse line (format: IP HOSTNAME)
+                    parts = line.split()
```
The host file syntax is a little bit more complex than just a split.
- Users might have multiple spaces or tabs en host files, we should support this use-case too
- There might be multiple hostname for the same IP, for example: `127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4` and this should be supported too

Dramelac commented on exegol/model/ContainerConfig.py:
```diff
@@ -1805,6 +1838,13 @@ def getTextPorts(self, is_running: bool = True) -> str:
+    def getTextExtraHosts(self) -> str:
+        """Text formatter for Extra Hosts configuration."""
+        result = ''
+        for hostname, ip in self.__extra_host.items():
```
We should exclude `self.hostname` to be displayed here, this extra env is set by default is Host network mode

Dramelac commented on exegol/model/ContainerConfig.py:
```diff
@@ -1374,6 +1376,37 @@ def getShellEnvs(self) -> List[str]:
+            logger.error(f"Hosts file not found: {hosts_file_path}")
+            return
+        if not hosts_path.is_file():
+            logger.error(f"Invalid hosts file path: {hosts_file_path}")
```
In case of user input error, maybe use `logger.critical` instead of `error` for both cases to avoid user creating a container without noticing the typo / config error in his command.

Dramelac reviewed (changes requested):
Hello !

Thank you for the PR! It was in the TODO for a long time.
Not much to change before merging

Bastian130 committed:
fix(config): improve hosts file parsing and error handling

- Support multiple spaces/tabs as separators using regex split
- Support multiple hostnames per IP (e.g., 127.0.0.1 localhost localhost4)
- Handle inline comments in hosts file entries
- Exclude container hostname from extra hosts display (auto-added in Host mode)
- Change logger.error to logger.critical for user input errors to prevent
  silent container creation with config typos

Bastian130 commented:
Hello ! 
Changes done :+1:

Dramelac committed:
Merge branch 'refs/heads/dev' into fork/Bastian130/feature/hosts-file-import

Dramelac committed:
Add extra host parsing + display hosts in container recap table

Dramelac reviewed (approved):
Thank you for your contribution !
